uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Service/Stats/StatsGatherer.php

```diff
@@ -263,4 +267,101 @@ private function isDrupal() {
     return $userSystem instanceof CRM_Utils_System_Drupal;
   }
 
+  /**
+   * Fetches entity counts for the job role entities
+   *
+   * @return array
+   */
+  private function getJobRoleEntityCounts() {
+    $recruitmentKey = 'com.civicrm.hrjobroles';
+    $counts = [];
+
+    if (!ExtensionHelper::isExtensionEnabled($recruitmentKey)) {
+      return $counts;
+    }
+
+    $counts['jobRole'] = $this->getEntityCount('HrJobRoles');
+    $params = ['option_group_id' => 'cost_centres'];
+    $counts['costCenter'] = $this->getEntityCount('OptionValue', $params);
+    $counts['funder'] = $this->getFunderCount();
+
+    return $counts;
+  }
+
+  /**
+   * Gets a count of all contacts in the system that have been marked as a
+   * funder
+   *
+   * @return int
+   */
+  private function getFunderCount() {
+    $result = civicrm_api3('HrJobRoles', 'get', ['return' => ['funder']]);
+    $result = array_column($result['values'], 'funder');
+    $funderIds = implode('', $result);
```

Comment:
I had a look on staging and it seems that, at least when there's only one funder, the `funder` value will not end with a delimiter. So, if you have a result that looks like this:

```
$result[
    ['funder' => 1],
    ['funder' => 2],
    ['funder' => 3],
    ['funder' => 4],
    ['funder' => 5],
    ['funder' => 6],
];
```

Imploding the values will result in something like this: `123456` and then the explode will not work

Reply:
That's interesting, I can see what you're talking about on staging too, but I tested this with multiple job roles (both with multiple funders and only single funders) and didn't see this.

On staging after adding a funder for an empty one:

`"funder": "|"` => `"funder": "|3|"`

And after editing one without a delimiter:

`"funder": "345"` => `"funder": "|345|"`

So it looks like delimiters are the expected approach. I'm not sure why staging has this kind of data or whether it's valid, any ideas? I can fix it anyway by using `|` as the glue for implode.

Reply:
I have no ideas why staging is like that, but it makes me wonder if there are any clients sites in the same situation. Using `|` as the glue is a good idea